src/axom/inlet/DocWriter.hpp

```diff
@@ -0,0 +1,35 @@
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "axom/sidre.hpp"
+
+#ifndef INLET_DOCWRITER_HPP
+#define INLET_DOCWRITER_HPP
+
+namespace axom
+{
+namespace inlet
+{ 
+
+class DocWriter {
+public: 
+  DocWriter(const std::string& fileName, axom::sidre::Group* sidreRootGroup, bool isVerbose = true);
+  void writeTitle(const std::string& title);
+  void writeSubtitle(const std::string& sub);
+  void writeTable(const std::string& title);
+
+  // Entry point for document generation
+  void writeDocuments(axom::sidre::Group* sidreGroup);
+
+private:
+  std::ofstream outFile;
+  axom::sidre::Group* sidreGroupRoot;

```

Comment:
For consistency with the rest of the component, can you rename this to sidreRootGroup